extractor/filesystem/containers/dockercomposeimage/dockercomposeimage.go

```diff
@@ -0,0 +1,206 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package dockercomposeimage extracts image urls from Docker Compose files.
+package dockercomposeimage
+
+import (
+	"context"
+	"errors"
+	"io"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/compose-spec/compose-go/v2/loader"
+	"github.com/compose-spec/compose-go/v2/types"
+	"github.com/google/osv-scalibr/extractor"
+	"github.com/google/osv-scalibr/extractor/filesystem"
+	"github.com/google/osv-scalibr/extractor/filesystem/internal/units"
+	"github.com/google/osv-scalibr/inventory"
+	"github.com/google/osv-scalibr/log"
+	"github.com/google/osv-scalibr/plugin"
+	"github.com/google/osv-scalibr/purl"
+	"github.com/google/osv-scalibr/stats"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	// Name is the unique name of this extractor.
+	Name = "containers/dockercomposeimage"
+
+	// DefaultMaxFileSizeBytes is the default maximum file size the extractor will
+	// attempt to extract. If a file is encountered that is larger than this
+	// limit, the file is ignored by `FileRequired`.
+	DefaultMaxFileSizeBytes = 1 * units.MiB
+)
+
+// Config is the configuration for the Extractor.
+type Config struct {
+	// Stats is a stats collector for reporting metrics.
+	Stats stats.Collector
+	// MaxFileSizeBytes is the maximum file size this extractor will unmarshal. If
+	// `FileRequired` gets a bigger file, it will return false,
+	MaxFileSizeBytes int64
+}
+
+// DefaultConfig returns the default configuration for the extractor.
+func DefaultConfig() Config {
+	return Config{
+		MaxFileSizeBytes: DefaultMaxFileSizeBytes,
+	}
+}
+
+// Extractor extracts repository URLs from Docker Compose files.
+type Extractor struct {
+	stats            stats.Collector
+	maxFileSizeBytes int64
+}
+
+// New returns a Docker Compose repository extractor.
+//
+// For most use cases, initialize with:
+// ```
+// e := New(DefaultConfig())
+// ```
+func New(cfg Config) *Extractor {
+	return &Extractor{
+		stats:            cfg.Stats,
+		maxFileSizeBytes: cfg.MaxFileSizeBytes,
+	}
+}
+
+// NewDefault returns an extractor with the default config settings.
+func NewDefault() filesystem.Extractor { return New(DefaultConfig()) }
+
+// Name of the extractor.
+func (e Extractor) Name() string { return Name }
+
+// Version of the extractor.
+func (e Extractor) Version() int { return 0 }
+
+// Requirements of the extractor.
+func (e Extractor) Requirements() *plugin.Capabilities { return &plugin.Capabilities{} }
+
+// FileRequired returns true if the specified file matches Docker Compose Configuration File.
+func (e Extractor) FileRequired(api filesystem.FileAPI) bool {
+	path := api.Path()
+	// Skip directories and oversized files
+	fi, err := os.Stat(path)
+	if err != nil || fi.IsDir() {
+		return false
+	}
+	if e.maxFileSizeBytes > 0 && fi.Size() > e.maxFileSizeBytes {
+		return false
+	}
+
+	// Parse YAML and look for top-level "services" field
```

Comment:
Prefer doing byte inspection in Extract() instead of FileRequired(). It's fine to return an empty inventory in these cases.